Add back orphaned file (#563)
SUSE/doc-susemanager: modules/salt/pages/salt-states.adoc

State data comes from {productname} in the form of package and custom states and targets minions at three specific levels of hierarchy.
The state hierarchy is defined by the following order or priority: individual minions have priority on packages and custom states over groups; next a group has priority over the organization.

* Minion Level
+
menu:Systems[Specific Minion > States]
* Group Level
+
menu:Systems[System Groups]
* Organization Level
+
menu:Systems[Manage System Types: > My Organization]

For example:

* Org1 requires that vim version 1 is installed
* Group1 requires that vim version 2 is installed
* Group2 requires any version installed

This would lead to the following order of hierarchy:

* Minion1 part of [Org1, Group1] wants vim removed, vim is removed (Minion Level)
* Minion2 part of [Org1, Group1] wants vim version 2 gets version 2 (Group Level)
* Minion3 part of [Org1, Group1] wants any version, gets version 2 (Org Level)
* Minion4 part of[Org1, Group2] wants any version, gets vim version 1 (Org Level)

[[bp.chapt.config.mgr.salt.states.storage.locations]]
== Salt States Storage Locations

> The {productname} salt-master reads its state data from three file root locations.

Reviewer: ```suggestion
The {productname} salt-master reads its state data from three directories.
```

Reviewer: maybe, file root is okay.  It is also used later on.

> The directory [path]``/usr/share/susemanager/salt``

Reviewer: ```suggestion
The directory [path]``/usr/share/susemanager/salt`` is used by {productname} and comes from the susemanager-sls.
```

Reviewer: what's susemanager-sls?  Is this a package name?  Either some kind of markup is missing or spelling has to be changed.

>  is used by {productname} and comes from the susemanager-sls.

Reviewer: ```suggestion
```

It is shipped and updated together with {productname} and includes certificate setup and common state logic to be applied to packages and channels.

The directory [path]``/srv/susemanager/salt`` is generated by {productname} and based on assigned channels and packages for minions, groups and organizations.
This file will be overwritten and regenerated.
This could be thought of as the {productname} database translated into salt directives.

> The third directory [path]``/srv/salt`` is for custom state data, modules etc.

Reviewer: ```suggestion
The third directory [path]``/srv/salt`` is for custom state data, modules, etc.
```

{productname} does not operate within or utilize this directory.

> However the state data placed here affects the Highstate of minions and is merged with the total state result generated by {productname}.

Reviewer: ```suggestion
However, the state data placed here affects the Highstate of minions and is merged with the total state result generated by {productname}.
```
see the other however sentence later on.

[[bp.chapt.config.mgr.susemgr.states]]
== {productname} States

> All sls files created by users will be saved to disk on the salt-master server.

Reviewer: ```suggestion
All user created SLS files will be saved to disk on the salt-master server.
```
Do we actually use salt-master?  Or Salt master.
I vote to cap SLS as you usually write PNG files, etc.

These files will be placed in [path]``/srv/susemanager/salt/``
 and each organization will be placed within its own directory.
Although these states are custom, these states are created using {productname}
.

> The following provides an overview of directory structure:

Reviewer: ```suggestion
The following provides an overview of the directory structure:
```

----
├── manager_org_DEVEL
│   ├── files
│   │    ... files needed by states (uploaded by users)...
│   └── state.sls
         ... other sls files (created by users)...
E.g.:
├── manager_org_TESTING
│   ├── files
│   │   └── motd     # user created
│   │    ... other files needed by states ...
│   └── motd.sls     # user created
            ... other sls files ...
----

[[bp.chapt.config.mgr.pillar.data]]
== Pillar Data

> SUSE Manager exposes a small amount of internal data as Pillars which can be used with custom {sls} states.

Reviewer: is {sls} the right macro here?

Author: Looks out of place, I think we can probably delete it entirely.

```suggestion
SUSE Manager exposes a small amount of internal data as Pillars which can be used with custom states.
```

Data that is exposed includes group membership, organization membership, and file roots.
These are managed either automatically by {productname}, or manually by the user.

> To avoid hard-coding organization IDs within {sls} files, a pillar entry is added for each organization:

Reviewer: Capitalizing pillar or not?

Author: I don't think we've capped it elsewhere.

----
org-files-dir: relative_path_to_files
----

The specified file is available for all minions which belong to the organization.

This is an example of a Pillar located at [path]``/etc/motd``:

----
file.managed:
    - source: salt://{{ pillar['org-files-dir']}}/motd
    - user: root
    - group: root
    - mode: 644
----

[[bp.sect.groupstates]]
== Group States

Pillar data can be used to perform bulk actions, like applying all assigned states to minions within the group.

> This section contains some example of bulk actions that you can take using group states.

Reviewer: execute?  Later we say perform (what I'd prefer here as well)

Author: Dunno. I think the whole thing needs a proper edit, it's old content.

In order to perform these actions, you will need to determine the ID of the group that you want to manipulate.
You can determine the Group ID by using the [command]``spacecmd`` command:

----
spacecmd group_details
----

In these examples we will use an example Group ID of [systemitem]``GID``.

To apply all states assigned to the group:

----
salt -I 'group_ids:GID' state.apply custom.group_GID
----

To apply any state (whether or not it is assigned to the group):

----
salt -I 'group_ids:GID' state.apply ``state``
----

To apply a custom state:

----
salt -I 'group_ids:2130' state.apply manager_org_1.``customstate``
----

Apply the highstate to all minions in the group:

----
salt -I 'group_ids:GID' state.apply
----

[[salt.pillars.setpackageendpoint]]

> == Use pillars to set the package download endpoint

Reviewer: ```suggestion
== Use Pillars to Set the Package Download Endpoint
```

By default, {productname} assumes that the download endpoint to use is the FQDN of the {productname} server, or the {productname} Proxy.
However, there are some cases where you might like to use a different FQDN as the download endpoint.
The most common example is if you need to use load balancing, caching proxies, or in environments with complicated networking requirements.

> To change the package download endpoint, you can manually adjust three salt pillars:

Reviewer: ```suggestion
To change the package download endpoint, you can manually adjust three Salt pillars:
```
